### PR TITLE
Make windows e2e tests run in parallel

### PR DESCRIPTION
**Pull request by KevinFairise2**
### What does this PR do?

Add some `t.Parallel()` on test that can benefit from it.
It allows us to speed some tests:
- new-e2e-windows-service-test: 1h10 -> 28 min
- new-e2e-windows-service-test-nofim: 50min -> 28min
- new-e2e-windows-systemprobe: 30min -> 18min
- new-e2e-installer-windows: 50 min -> 33 min

### Motivation

Increase CI speed, without increasing resource consumption

### CODE REVIEW

File: test/new-e2e/tests/security-agent-functional/security_agent_test.go [hunk 1]
```diff
@@ -36,6 +36,7 @@ var (
 
 func TestVMSuite(t *testing.T) {
 	flake.Mark(t)
+	t.Parallel()
```
Comment (paulcacheux) [inline test/new-e2e/tests/security-agent-functional/security_agent_test.go:39]: I'm not sure I see the point, this suite has only one test anyway.

Side question: this test is marked as flaky currently, was this tested without the marker ?

**Review by paulcacheux — CHANGES_REQUESTED** [review]

File: test/new-e2e/tests/security-agent-functional/security_agent_test.go (hunk 1 above)
Comment (KevinFairise2) [reply test/new-e2e/tests/security-agent-functional/security_agent_test.go:39]: You're right I should not have kept it if it is the single test in the package. Removing

**Review by paulcacheux — APPROVED** [review]

**Review by clarkb7 — CHANGES_REQUESTED** [review]
`new-e2e-windows-service-test` was just parallelized yesterday in https://github.com/DataDog/datadog-agent/pull/31334
That same PR also removes the `-nofim` test.

Since it was done with  `parallel.matrix` there's no PR conflicts, tricky.

That still leaves the `installer-windows` tests. I'm generally opposed to using `t.Parallel()` for test parallelization because it makes the job logs unreadable, which was a huge issue with the kitchen tests.

**Comment by KevinFairise2** [comment]
> `new-e2e-windows-service-test` was just parallelized yesterday in #31334 That same PR also removes the `-nofim` test.
> 
> Since it was done with `parallel.matrix` there's no PR conflicts, tricky.
> 
> That still leaves the `installer-windows` tests. I'm generally opposed to using `t.Parallel()` for test parallelization because it makes the job logs unreadable, which was a huge issue with the kitchen tests.

I must admit that all the logs are displayed in the same job. It makes is less readable but should still be readable because each log section should be prefixed by `=== RUN   <test name>`.

Thanks for pointing out the `new-e2e-windows-service-test` updates because I indeed did not notice them! If you absolutely want to avoid `t.Parallel()` I can remove the windows-installer part from the PR. 
But could you please open another PR to make then run in parallel with a matrix? These jobs are in the critical path and any improvement on their duration makes the overall pipeline faster.

**Review by derekwbrown — CHANGES_REQUESTED** [review]
I'm going to have to second Branden's comment.

IIUC, this will make the logs interlacesd; if we have a test to debug, it will make it very difficult.

If you're going to parellelize, I would suggest either
a) some toolling that post processes the logs such that they're readable, separated by test or
b) use the parallel.matrix technique branden described.

**Comment by alexn-dd** [comment]
One important note is that, if some Windows tests make use of COM, spinning threads can potentially break COM specific requirements, leading to ambiguous failures.

**Comment by pducolin** [comment]
Hi! Here to confirm what @KevinFairise2 says: we have all other teams running their tests in parallel leveraging `t.Parallel()`.

Indeed no risk of COM overlapping, I see some confusion here between the test runner, that is a k8s linux pod on `dd-build` and the device under test. The test runner is the one that runs the go test and that leverages the `t.Parallel` feature. Each test has its own EC2, sandboxed instance. I recommend [reading our doc](https://datadoghq.atlassian.net/wiki/spaces/ADX/pages/3492282435/E2E+Overview) to learn more about how e2e tests work 

As @KevinFairise2 said we worked on improving logs readability to allow reducing noise. For instance lower logs such as pulumi's are printed only on error.

The cons of parallel matrix is that it creates one test runner per test, which increases costs: we need to wait for each runner to be up, plus we use x runners while one could be enough.

I invite you to take a look at this example of runner running tests in parallel leveraging `t.Parallel` 

https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/720043578

As you can see, as @KevinFairise2 said, each test runs and prints its log lines prefixed by 

```
=== RUN   <TestName>
....
```

**Pull request closed without merging**